src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.NormalSearch.cs

```diff
@@ -41,14 +40,21 @@ internal abstract partial class AbstractNavigateToSearchService
             return;
         }
 
-        await SearchDocumentInCurrentProcessAsync(document, searchPattern, kinds, onItemFound, cancellationToken).ConfigureAwait(false);
+        await SearchDocumentInCurrentProcessAsync(document, searchPattern, kinds, onItemsFound, cancellationToken).ConfigureAwait(false);
     }
 
-    public static Task SearchDocumentInCurrentProcessAsync(Document document, string searchPattern, IImmutableSet<string> kinds, Func<RoslynNavigateToItem, Task> onItemFound, CancellationToken cancellationToken)
+    public static async Task SearchDocumentInCurrentProcessAsync(
+        Document document, string searchPattern, IImmutableSet<string> kinds, Func<ImmutableArray<RoslynNavigateToItem>, Task> onItemsFound, CancellationToken cancellationToken)
     {
-        return SearchProjectInCurrentProcessAsync(
-            document.Project, priorityDocuments: [], document, searchPattern, kinds,
-            onItemFound, () => Task.CompletedTask, cancellationToken);
+        var (patternName, patternContainerOpt) = PatternMatcher.GetNameAndContainer(searchPattern);
+        var declaredSymbolInfoKindsSet = new DeclaredSymbolInfoKindSet(kinds);
+
+        var results = new ConcurrentSet<RoslynNavigateToItem>();
+        await SearchSingleDocumentAsync(
+            document, patternName, patternContainerOpt, declaredSymbolInfoKindsSet, t => results.Add(t), cancellationToken).ConfigureAwait(false);
```

Comment:
>t => results.Add(t) [](http://example.com/codeflow?start=84&length=19)

StandardCallbacks<RoslynNavigateToItem>.AddToConcurrentSet?

Reply:
would require passing around the set though

Reply:
yeah.  don't care  here :)